pkg/dispatchable/dispatchable.go

```diff
@@ -0,0 +1,65 @@
+package dispatchable
+
+import (
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+func NewDispatchFinder(l hclog.Logger, sliceAtoms []types.Atom) *DispatchFinder {
+	mAtoms := make(map[types.SpecTuple]types.Atom)
```

Comment:
You need mutexes around this map to prevent concurrent writes.  As written this can panic if a concurrent read and write try to touch this map from different goroutines.